---
title: 'sam file field SEQ'
date: 2020-07-04
permalink: /posts/2020/07/03/blog
tags:
  - Bioinformatics
  - Sam
  - Bowtie2
---

In [Samtools manual](https://samtools.github.io/hts-specs/SAMv1.pdf "Samtools manual"), the SEQ field is not fully defined. 

> SEQ: segment SEQuence.  This field can be a ‘ \* ’  when the sequence is not stored.  If not a ‘*’, the lengthof the sequence must equal the sum of lengths ofM/I/S/=/Xoperations inCIGAR. An ‘=’ denotes thebase is identical to the reference base.  No assumptions can be made on the letter cases. 

------------
Bowtie2 will print read sequence in this field.

if aligned to the reverse strand, reverse-complemented reads sequence will be printed.

Tested, confirmed.

------------

BWA probably does it in the same way.

> 10	SEQ	query SEQuence on the same strand as the reference

No test yet

------------

Samtools fasta/fastq
It dumps SEQ field but in Raw read style. Essentially, raw reads will be output.

For example:

A sequence with flag bit 16, means it is mapped to reverse strand. In SEQ field, a reverse complement sequence is stored in SEQ field. However, when outputing in samtools fasta, original sequence will be written out.

